clang/lib/Driver/ToolChain.cpp

```diff
@@ -1169,7 +1169,7 @@ llvm::opt::DerivedArgList *ToolChain::TranslateOffloadTargetArgs(
     // matches the current toolchain triple. If it is not present
     // at all, target and host share a toolchain.
     if (A->getOption().matches(options::OPT_m_Group)) {
-      if (SameTripleAsHost)
+      if (SameTripleAsHost || getTriple().getArch() == llvm::Triple::amdgcn)
```

Comment:
Could you please clarify why the AMD GPU device target goes down the same path as host here, possibly by extending the comment before this `if`-block? Sorry if it's something that I'm missing in the above discussion of other components' intricacies.

Reply:
`mcpu` option is necessary for build AMD GPU device code. This original code will filters out the this option even if the user specifies the `mcpu` option on the command line. Maybe there is a better solution?